cachito/web/api_v1.py

```diff
@@ -588,3 +590,41 @@ def get_request_logs(request_id):
     return flask.Response(
         stream_with_context(generate_stream_response(log_file_path)), mimetype="text/plain"
     )
+
+
+@api_v1.route("/content-manifest", methods=["GET"])
+def get_content_manifest_by_requests():
+    """
+    Retrieve the content manifest associated with the given requests.
+
+    :return: a Flask JSON response
+    :rtype: flask.Response
+    :raise NotFound: if one of the given request is not found
+    :raise BadRequest: if any of the given request is not in the "complete" or "stale" state
+    """
+    arg = flask.request.args.get("requests")
+    if not arg:
+        return flask.jsonify(BASE_ICM)
+    request_ids = []
+    item: str
+    for item in arg.split(","):
+        if not item.strip():
+            continue
+        if not item.strip().isdigit():
+            raise BadRequest(f"{item} is not an integer.")
+        request_ids.append(int(item))
+
+    requests = Request.query.filter(Request.id.in_(request_ids))
+    assembled_icm = deepcopy(BASE_ICM)
+    request: Request
+    for request in requests:
+        if request.state.state_name not in (
+            RequestStateMapping.complete.name,
+            RequestStateMapping.stale.name,
+        ):
+            raise BadRequest(
+                f"Request {request.id} is in state {request.state.state_name}, "
+                f"not complete or stale."
+            )
+        assembled_icm["image_contents"].extend(request.content_manifest.to_json()["image_contents"])
```

Comment:
@tkdchen as we discussed, I've been thinking about removing duplicates and unfortunately, I don't think de-duplicating by `purl` would be enough. There are various cases in which different packages can have the same purl.

For example, if your repo contains both a `pip` package and an `npm` package in the root directory, their purls will both be `pkg:github/<repo>@<ref>`.
Other example, if two different repos contain gomod packages with exactly the same name and version, their purls will also be the same. This *shouldn't* happen, but nothing actually prevents the users from doing it.
Both of these are cases are bad and Cachito should handle them better, but purls in Cachito are kind of a mess at the moment.

We can remove duplicates only if all the attributes (`purl`, `dependencies`, `sources`) are exactly the same. Maybe that would be worth doing?